runtime/vm/jvminit.c

```diff
@@ -5443,8 +5443,10 @@ protectedInitializeJavaVM(J9PortLibrary* portLibrary, void * userData)
 	 * 			This applies to Windows only for now. Since we don't have Hursley JCLs for this, we will need to provide our own support for all other cases/platforms.
 	 */
 	if (J9_ARE_NO_BITS_SET(vm->sigFlags, J9_SIG_XRS_ASYNC)) {
-		/* only register the handler if -Xrs is not present */
-		if (0 != j9sig_set_async_signal_handler(shutDownHookWrapper, vm, J9PORT_SIG_FLAG_SIGTERM)) {
+		/* Only register the handler if -Xrs is not present. This is a temporary hack,
+		 * which will be removed once the required OMR signal API has been implemented.
+		 */
+		if (0 != j9sig_set_async_signal_handler(shutDownHookWrapper, vm, J9PORT_SIG_FLAG_SIGTERM | J9PORT_SIG_FLAG_SIGINT)) {
```

Comment:
Not sure if we should accept this "hack", but if so it needs to be #ifdef'ed so it only affects OpenJ9 builds. Also the comment should be clear about which part should be removed later.

Reply:
Is there an issue # for the OMR signal API implementation?  That will make it easier to know which issue needs to be closed to complete this.

Reply:
everything should be removed later. even code for `J9PORT_SIG_FLAG_SIGTERM` is a hack. the `shutDownHookWrapper` registered here gets overridden when [Terminator.java](https://github.ibm.com/runtimes/openj9-openjdk-jdk9/blob/ibm_sdk/jdk/src/java.base/windows/classes/java/lang/Terminator.java#L61) registers handlers for `SIGINT` and `SIGTERM` using `sun.misc.Signal.register`. this code shouldn't influence IBM J9's behavior.

Reply:
> Is there an issue # for the OMR signal API implementation?

Referred to OMR issue eclipse/omr#2332 in the code and pull request.